Add support for address labels in extended format

ozjonesy opened the pull request:
Address labels were not working and causing the deserialisation to drop a big chunk of the config.

- Update AddressMapping enum to support both simple and extended formats
- Modify AddressProperties struct to include optional label field
- Ensure backwards compatibility with simple address format
- Fixes issue where addresses with labels were being dropped during deserialisation

NeonD-Luke committed:
Add support for address labels in extended format

- Update AddressMapping enum to support both simple and extended formats
- Modify AddressProperties struct to include optional label field
- Ensure backwards compatibility with simple address format
- Fixes issue where addresses with labels were being dropped during deserialisation

TobiasDeBruijn commented:
Good catch. Thanks!

I cannot find the `preferred_lifetime` option in the [Netplan docs](https://netplan.readthedocs.io/en/stable/netplan-yaml/#properties-for-all-device-types), other than in the documentation for the `lifetime` field. Where'd that one come from?

Edit: If you have the time, would you mind adding some unit tests as well? I know it's not done anywhere yet, but it's never too late to start I suppose.

NeonD-Luke committed:
Updated to remove PreferredLifetime and added some unit tests

ozjonesy commented:
Ha, sorry co-pilot added it in and I didn't check it well enough. I was more focused on label.
I have updated it and added some unit tests. Hope that helps.

TobiasDeBruijn committed:
Fix: Use PreferredLifetime instead of String
